src/main/drivers/pwm_output.c

```diff
@@ -373,6 +385,125 @@ uint32_t getDshotHz(motorPwmProtocolTypes_e pwmProtocolType)
         return MOTOR_DSHOT150_HZ;
     }
 }
+
+void pwmWriteDshotCommand(uint8_t index, uint8_t motorCount, uint8_t command, bool blocking)
+{
+    timeUs_t timeNowUs = micros();
+    timeUs_t timeDelayUs;
+    if (isMotorProtocolDshot() && (command <= DSHOT_MAX_COMMAND)) {
+        uint8_t repeats;
+        switch (command) {
+        case DSHOT_CMD_SPIN_DIRECTION_1:
+        case DSHOT_CMD_SPIN_DIRECTION_2:
+        case DSHOT_CMD_3D_MODE_OFF:
+        case DSHOT_CMD_3D_MODE_ON:
+        case DSHOT_CMD_SAVE_SETTINGS:
+        case DSHOT_CMD_SPIN_DIRECTION_NORMAL:
+        case DSHOT_CMD_SPIN_DIRECTION_REVERSED:
+            repeats = 10;
+            timeDelayUs = DSHOT_COMMAND_DELAY_US;
+            break;
+        case DSHOT_CMD_BEACON1:
+        case DSHOT_CMD_BEACON2:
+        case DSHOT_CMD_BEACON3:
+        case DSHOT_CMD_BEACON4:
+        case DSHOT_CMD_BEACON5:
+            repeats = 1;
+            timeDelayUs = DSHOT_BEEP_DELAY_US;
+            break;
+        default:
+            repeats = 1;
+            timeDelayUs = DSHOT_COMMAND_DELAY_US;
+
+            break;
+        }
+        if (blocking) {
+            for (; repeats; repeats--) {
+                for (uint8_t i = 0; i < motorCount; i++) {
+                    delayMicroseconds(DSHOT_COMMAND_DELAY_US);
+                    if ((i == index) || (index == ALL_MOTORS)) {
+                        motorDmaOutput_t *const motor = getMotorDmaOutput(i);
+                        motor->requestTelemetry = true;
+                        pwmWriteDshotInt(i, command);
+                    }
+                }
+
+                pwmCompleteDshotMotorUpdate(0);
+            }
+            delayMicroseconds(timeDelayUs);
+        } else {
+            for (uint8_t i = 0; i < motorCount; i++) {
+                if ((i == index) || (index == ALL_MOTORS)) {
+                    if (dshotCommandControl.command[i] == 0) {
+                        dshotCommandControl.command[i] = command;
+                        dshotCommandControl.repeats = repeats;
+                        dshotCommandControl.nextCommandAt = timeNowUs + DSHOT_COMMAND_DELAY_US;
+                        dshotCommandControl.delayAfterCommand = timeDelayUs;
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+bool pwmIsDshotCommand(void)
+{
+    if (dshotCommandControl.nextCommandAt) {
+        return(true);
```

Comment:
`return` is an operator and not a function

Reply:
Yup you right. I'll fix those